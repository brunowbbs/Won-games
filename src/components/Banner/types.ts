import { RibbonSizes } from "../Ribbon/types";

export type BannerProps = {
  img: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
  ribbon: React.ReactNode;
  ribbonSize: RibbonSizes;
  ribbonCollor: "primary" | "secondary";
};
