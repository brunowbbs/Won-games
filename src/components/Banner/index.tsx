import Button from "../Button";
import Ribbon from "../Ribbon";
import * as S from "./styles";
import { BannerProps } from "./types";

const Banner = ({
  buttonLabel,
  buttonLink,
  img,
  subtitle,
  title,
  ribbon,
  ribbonCollor = "primary",
  ribbonSize = "normal",
}: BannerProps) => (
  <S.Wrapper>
    {!!ribbon && (
      <Ribbon color={ribbonCollor} size={ribbonSize}>
        {ribbon}
      </Ribbon>
    )}

    <S.Image src={img} role="img" aria-label={title} />
    <S.Caption>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink} size="large">
        {buttonLabel}
      </Button>
    </S.Caption>
  </S.Wrapper>
);

export default Banner;
