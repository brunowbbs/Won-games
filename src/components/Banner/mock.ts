export const bannerMock = {
  img: "https://source.unsplash.com/user/willianjusten/1042x580",
  title: "Defy death",
  subtitle: "Play the new CrashLand season",
  buttonLabel: "Buy now",
  buttonLink: "/games/defy-death",
};

export const bannersMock = [
  {
    img: "https://source.unsplash.com/user/willianjusten/1042x580",
    title: "Defy death 1",
    subtitle: "Play the new CrashLand season",
    buttonLabel: "Buy now",
    buttonLink: "/games/defy-death",
  },
  {
    img: "https://source.unsplash.com/user/willianjusten/1042x582",
    title: "Defy death 2",
    subtitle: "Play the new CrashLand season",
    buttonLabel: "Buy now",
    buttonLink: "/games/defy-death",
  },
  {
    img: "https://source.unsplash.com/user/willianjusten/1042x581",
    title: "Defy death 3",
    subtitle: "Play the new CrashLand season",
    buttonLabel: "Buy now",
    buttonLink: "/games/defy-death",
  },
];
