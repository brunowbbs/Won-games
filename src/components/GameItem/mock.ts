export const gameItemMock = {
  img: "https://source.unsplash.com/user/willianjusten/151x70",
  title: "Red Dead Redemption 2",
  price: "R$ 215,00",
  paymentInfo: {
    flag: "mastercard",
    img: "/img/master-card.png",
    number: "**** **** **** 4326",
    purchaseDate: "Purchase made on 07/20/2020 at 20:32",
  },
};
