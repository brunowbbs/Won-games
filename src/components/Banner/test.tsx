import { screen } from "@testing-library/react";

import { renderWithTheme } from "../../utils/tests/helpers";

import Banner from ".";
import { bannerMock } from "./mock";

describe("<Banner/>", () => {
  it("Should render correctly", () => {
    const { container } = renderWithTheme(<Banner {...bannerMock} />);

    expect(screen.getByRole("heading", { name: /Defy death/i }));
    expect(
      screen.getByRole("heading", { name: /Play the new CrashLand season/i })
    );
    expect(
      screen.getByRole("img", { name: /Defy death/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
