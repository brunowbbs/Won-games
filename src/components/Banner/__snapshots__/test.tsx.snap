// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> Should render correctly 1`] = `
<div
  class="sc-dkrFOg ceMuHe"
>
  <div
    aria-label="Defy death"
    class="sc-hLBbgP iroKqw"
    role="img"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class="sc-eDvSVe kmYFIO"
  >
    <h2
      class="sc-jSUZER jrYSWl"
    >
      Defy death
    </h2>
    <h3
      class="sc-gKPRtg hGHUMN"
    >
      Play the new CrashLand season
    </h3>
    <a
      class="sc-bcXHqe jsimA"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</div>
`;
